src/components/listings/add/AddListing.js

```diff
@@ -0,0 +1,283 @@
+        } catch (error) {
+          setSubmitMessage(error.response.data.data.errorMessage);
+        }
+      };
+
+      submitForm();
+    }
+  };
+
+  return (
+    <section id="add-listings">
+      <h1>Add a Listing</h1>
+
+      <form id="add-listings-form">
+        <div className="form-input-container">
+          <label className="form-input-label">Name*</label>
+          <input
+            name="name"
+            type="text"
+            className={`form-input ${
+              dataErrors.nameError ? "form-input-error" : ""
+            }`}
+            onChange={ handleInputChange }
+          />
+          <span>{ dataErrors.nameError }</span>
+        </div>
+
+        <div className="form-input-container">
+          <label className="form-input-label">Phone*</label>
+          <input
+            name="phone"
+            type="tel"
+            className={`form-input ${
+              dataErrors.phoneError
+                ? dataErrors.phoneError === "Invalid phone number"
+                  ? "form-input-warning"
+                  : "form-input-error"
+                : ""
+            }`}
+            onChange={ handleInputChange }
+          />
+          <span>{ dataErrors.phoneError }</span>
+        </div>
+
+        <div className="form-input-container">
+          <label className="form-input-label">Location*</label>
+          <input
+            name="location"
+            type="text"
+            className={`form-input ${
+              dataErrors.locationError ? "form-input-error" : ""
+            }`}
+            onChange={ handleInputChange }
+          />
+          <span>{ dataErrors.locationError }</span>
+        </div>
+
+        <div className="form-input-container">
+          <label className="form-input-label">Medication name*</label>
+          <input
+            name="medication_name"
+            type="text"
+            className={`form-input ${
+              dataErrors.medicationNameError ? "form-input-error" : ""
+            }`}
+            onChange={ handleInputChange }
+          />
+          <span>{ dataErrors.medicationNameError }</span>
+        </div>
+
+        <div className="form-input-container">
+          <label className="form-input-label">Quantity*</label>
+          <input
+            name="quantity"
+            type="number"
+            className={`form-input ${
+              dataErrors.quantityError ? "form-input-error" : ""
+            }`}
+            onChange={ handleInputChange }
+            min="1"
+          />
+          <span>{ dataErrors.quantityError }</span>
+        </div>
+
+        <div className="form-input-container">
+          <label className="form-input-label">Needed By*</label>
+          <input
+            name="needed_by"
+            type="date"
+            className={`form-input ${
+              dataErrors.dateError
+                ? dataErrors.dateError === "Date must be greater than today"
```

Comment:
Date could be today. Many people want the medication when they need it. So it should be "Cannot choose a past date"

Reply:
Ok, please merge, this is to be fixed in the localization json file.

Reply:
Why there though?

Reply:
It's coming from the backend.